lib/opentelemetry_plug.ex

```diff
@@ -27,34 +27,40 @@ defmodule OpentelemetryPlug do
     # register the tracer. just re-registers if called for multiple repos
     _ = OpenTelemetry.register_application_tracer(:opentelemetry_plug)
 
+    :telemetry.attach({__MODULE__, {event_prefix, :opentelemetry_phoenix_tracer_router_start}},
+      [:phoenix, :router_dispatch, :start],
+      &__MODULE__.handle_route/4,
+      config)
+
+    :telemetry.attach({__MODULE__, {event_prefix, :opentelemetry_plug_tracer_router_start}},
+      [:plug, :router_dispatch, :start],
+      &__MODULE__.handle_route/4,
+      config)
+
     :telemetry.attach({__MODULE__, {event_prefix, :opentelemetry_plug_tracer_start}},
-      event_prefix ++ [:start],
+      [:plug_adapter, :call, :start],
       &__MODULE__.handle_start/4,
       config)
 
     :telemetry.attach({__MODULE__, {event_prefix, :opentelemetry_plug_tracer_stop}},
-      event_prefix ++ [:stop],
+      [:plug_adapter, :call, :stop],
       &__MODULE__.handle_stop/4,
       config)
 
     :telemetry.attach({__MODULE__, {event_prefix, :opentelemetry_plug_tracer_exception}},
-      event_prefix ++ [:exception],
+      [:plug_adapter, :call, :exception],
       &__MODULE__.handle_exception/4,
       config)
   end
```

Comment:
Yea, think I wasn't sure if it would be used for anything else, like adding support for Plug.Telemetry as well. But will remove it until that is added, if ever.